pkg/diffs/diffs.go

```diff
@@ -0,0 +1,66 @@
+package diffs
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	prowconfig "k8s.io/test-infra/prow/config"
+)
+
+// Diffs holds the information where the differences can be found.
+type Diffs struct {
+	ConfigPath     string
+	JobConfigPath  string
+	DecorationPath string
+}
+
+// NewDiffs creates a new Diffs{}
+func NewDiffs(configPath, jobConfigPath, decorationPath string) *Diffs {
+	return &Diffs{
+		ConfigPath:     configPath,
+		JobConfigPath:  jobConfigPath,
+		DecorationPath: decorationPath,
+	}
+}
+
+// GetPresubmitsToExecute returns a mapping of repo to presubmits to execute.
+func (d *Diffs) GetPresubmitsToExecute() (map[string][]prowconfig.Presubmit, error) {
```

Comment:
I think do not assume what the Presubmits are for. They are Presubmits affected by a proposed change, what they are used for is not this code's business. The ...ToExecute suffix increases the code coupling unnecessarily.